### PR TITLE
Make scrolling with a touchpad in activity list and search result more natural

### PR DESCRIPTION
**Pull request by CarlSchwan**
This basically uses the same method that is used in Kirigami and Plasma
Components3.

### CODE REVIEW

File: src/gui/tray/wheelhandler.cpp [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+////////////////////////////
+KirigamiWheelEvent::KirigamiWheelEvent(QObject *parent)
+    : QObject(parent)
+{}
+
+KirigamiWheelEvent::~KirigamiWheelEvent() = default;
+
+void KirigamiWheelEvent::initializeFromEvent(QWheelEvent *event)
+{
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
```
Comment (mgallien) [inline src/gui/tray/wheelhandler.cpp:192]: we do already require Qt 5.15

File: src/gui/tray/wheelhandler.h [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+     * Whether the delta values are inverted
+     * On some platformsthe returned delta are inverted, so positive values would mean bottom/left
+     */
+    Q_PROPERTY(bool inverted READ inverted CONSTANT)
+
+    /**
+     * accepted: bool
+     *
+     * If set, the event shouldn't be managed anymore,
+     * for instance it can be used to block the handler to manage the scroll of a view on some scenarions
```
Comment (mgallien) [inline src/gui/tray/wheelhandler.h:87]: there is a typo on `scenarions`

**Review by mgallien — COMMENTED** [review]
thanks,
some small comments

**Review by mgallien — APPROVED** [review]

File: src/gui/tray/wheelhandler.cpp [hunk 3]
```diff
@@ -0,0 +1,280 @@
+/*
```
Comment (FlexW) [inline src/gui/tray/wheelhandler.cpp:1]: Should we not pack that into the third party folder?

File: src/gui/tray/wheelhandler.h [hunk 4]
```diff
@@ -0,0 +1,213 @@
+/*
```
Comment (FlexW) [inline src/gui/tray/wheelhandler.h:1]: same for this file.

File: src/gui/tray/wheelhandler.cpp (hunk 3 above)
Comment (FlexW) [reply src/gui/tray/wheelhandler.cpp:1]: If not, I would tend to say we should adjust the naming to make it more consistent with the current code base.

Comment (CarlSchwan) [reply src/gui/tray/wheelhandler.cpp:1]: Moved to the 3rdparty repo

**Review by FlexW — APPROVED** [review]
Very nice. Thanks :)

**Pull request merged**